miner.c

```diff
@@ -10508,8 +10505,11 @@ enum test_nonce2_result _test_nonce2(struct work *work, uint32_t nonce, bool che
 			{
 				applog(LOG_DEBUG, "Stratum pool %u target has changed since work job issued, checking that too",
 				       pool->pool_no);
-				if (hash_target_check_v(work->hash, pool->next_target))
+				if (hash_target_check_v(work->hash, pool->next_target)) {
 					high_hash = false;
+					memcpy(work->target, pool->next_target, sizeof(work->target));
+					calc_diff(work, 0);
```

Comment:
1) Agree with making it a pool option, will do;
2) I'd made it enabled by default only to keep the current behaviour unchanged. If you are ok with changing the default behaviour, I am ok too;
3) Maybe they are not being ignored anymore as a side effect of my change, I can't tell. But even if BFGMiner ignores them, pool does not and user still gets nasty high reject counters on the pool side. So: a) pool is broken and then there are no rejects, or b) pool is not broken and then even if BFGMiner ignores rejects, they are still being counted on pool side. I would say, there is no point in ignoring them at all.